src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
```

Comment:
Instead of checking for the inclusion of 'YYYY' and 'YY' in both `fromFormat` and `toFormat` separately, you could create a more generic conversion function that handles the year conversion. This would make the code easier to understand and maintain, especially if more year formats are introduced in the future.